sentinelhub/sh_utils.py

```diff
@@ -1,40 +1,102 @@
 """
 Module implementing some utility functions not suitable for other utility modules
 """
+from abc import ABC, abstractmethod
 from urllib.parse import urlencode
 
-from sentinelhub.config import SHConfig
-from sentinelhub.download import SentinelHubDownloadClient
 from sentinelhub.exceptions import MissingDataInRequestException
 
 
-def iter_pages(service_url, client=None, config=None, exception_message=None, **params):
-    """ Iterates over pages of items retrieved from Sentinel-Hub service
+class FeatureIterator(ABC):
```

Comment:
General comment about `FeatureIterators`... Is it possible to have any tests that are not "integration tests"?

Reply:
Sure, I can implement a dummy `_fetch_features` method and prepare a few tests. Will do in the morning.

Reply:
Added some unit tests that verify the basic mechanics of `FeatureIterator` class is working correctly.